client/src/components/IssueFileInput/index.tsx

```diff
@@ -13,80 +11,90 @@ import {
 } from 'constants/FileType';
 
 import { useTranslation } from 'react-i18next';
+import { getFnameByLink } from 'helpers/getFnameByLink';
 
 interface Props {
-	onChangePending?: (pending: boolean) => void;
-	onChange?: (newLinks: string[]) => void;
-	currentLinks?: string[];
-	issueKey: string;
+	currentFiles?: File[];
+	alreadyAttached?: string[];
+	onChange?: (newFiles: File[]) => void;
+	onDeleteAlreadyAttached?: (newLinks: string[]) => void;
 }
 
 const IssueFileInput: React.FC<Props> = ({
 	onChange = () => {},
-	onChangePending = () => {},
-	currentLinks: givenCurrentLinks,
-	issueKey,
+	currentFiles: givenCurrentFiles,
+	alreadyAttached = [],
+	onDeleteAlreadyAttached = () => {},
 }) => {
-	const currentLinks = givenCurrentLinks ?? [];
+	const currentFiles = givenCurrentFiles ?? [];
 	const { t } = useTranslation();
 
-	const upload = async (newFileList: File[]): Promise<string[]> => {
-		return await Promise.all(newFileList.map((file) => attachFile(file, issueKey)));
-	};
-
-	const handleUpload = (newFileList: FileList | null) => {
+	const handleChange = (newFileList: FileList | null) => {
 		if (newFileList) {
-			onChangePending(true);
-
-			upload(
-				Array.from(newFileList).filter((file) => ALLOWED_ISSUE_ATTACHMENT_MIME_TYPES.includes(file.type)),
-			).then((newLinks) => {
-				onChangePending(false);
-				onChange([...currentLinks, ...newLinks]);
-			});
+			onChange([
+				...currentFiles,
+				...Array.from(newFileList).filter((file) => ALLOWED_ISSUE_ATTACHMENT_MIME_TYPES.includes(file.type)),
+			]);
 		}
 	};
 
-	const curryDeleteLink = (currentLink: string) => () => {
-		const index = currentLinks.findIndex((linkToCheck) => linkToCheck === currentLink);
-		const newLinks = [...currentLinks];
+	const curryDeleteFile = (index: number) => () => {
+		const newFiles = [...currentFiles];
+		newFiles.splice(index, 1);
+		onChange(newFiles);
+	};
+
+	const curryDeleteLink = (index: number) => () => {
+		const newLinks = [...alreadyAttached];
 		newLinks.splice(index, 1);
-		onChange(newLinks);
+		onDeleteAlreadyAttached(newLinks);
+	};
+
+	const Body: React.FC = () => {
```

Comment:
move it to separate component

Reply:
I can't it uses many props and functions from IssueFileInput
I will move it to render